packages/data-stores/src/queries/use-site-domains-query.ts

```diff
@@ -96,7 +111,17 @@ export function useSiteDomainsQuery(
 	siteIdOrSlug: number | string | null | undefined,
 	options: UseQueryOptions< SiteDomainsQueryFnData > = {}
 ) {
-	return useQuery( getSiteDomainsQueryObject( siteIdOrSlug, options ) );
+	const transformedQuery = useQuery( getSiteDomainsQueryObject( siteIdOrSlug, options ) );
```

Comment:
@p-jackson The intention here is to transform the response into the `ResponseDomain` object. Maybe there's a better way of doing that here.

Reply:
IMO the queries in `@automattic/data-stores` should just be thin wrappers around the API. Just a common place we can define types, queries, and ensure requests are using a shared cache. So I think things like converting the domains into `ResponseDomain` objects is something that should happen in `@automattic/domains-table`; it's something that only our particular use case cares about. If we add it to `@automattic/data-stores` maybe other people will end up depending on all of the extra logic we had to bring over from Calypso, when in reality all they needed was a quick way to query the domains API.

So because of that I think we should keep `useSiteDomainsQuery` mostly as it is, and then pass down a `select` option, which is React Queries way to transform the data. It requires the types to be a little bit fancier, but to be fair I probably should have written them this way to begin with.

The type signature for `useSiteDomainsQuery` turns into:

```ts
export function useSiteDomainsQuery< TError = unknown, TData = SiteDomainsQueryFnData >(
	siteIdOrSlug: number | string | null | undefined,
	options: UseQueryOptions< SiteDomainsQueryFnData, TError, TData > = {}
) {
	// ...
}
```

And the type signature for `getSiteDomainsQueryObject` turns into:

```ts
export function getSiteDomainsQueryObject< TError = unknown, TData = SiteDomainsQueryFnData >(
	siteIdOrSlug: number | string | null | undefined,
	options: UseQueryOptions< SiteDomainsQueryFnData, TError, TData > = {}
): UseQueryOptions< SiteDomainsQueryFnData, TError, TData > {
	// ...
}
```

Those changes make the functions generic enough so they can be called with a different `select` option. So now in `<DomainsTableRow>` when we call `useSiteDomainsQuery` we can do this:

```tsx
	const { data: allSiteDomains, isLoading: isLoadingSiteDomainsDetails } = useSiteDomainsQuery(
		domain.blog_id,
		{
			enabled: inView,
			...( fetchSiteDomains && { queryFn: () => fetchSiteDomains( domain.blog_id ) } ),
			// This is the new bit:
			select: ( state ) => state.domains.map( createSiteDomainObject ),
		}
	);
```

Now `allSiteDomains` is a `ResponseDomain[]`, we don't even need the wrapping object because we don't care about it in the domains table.

And using `select` doesn't mess up the cache for other users of `useSiteDomainsQuery` because [it doesn't change what's stored in the query cache](https://tanstack.com/query/v4/docs/react/reference/useQuery).

Reply:
You could also make a `@automattic/domains-table` specific hook if there were multiple places where you wanted to query for site domains and then convert them into `ResponseData`s. Like how the `useDomainsTable` wraps `useAllDomainsQuery` to do table-specific things with the data.

Reply:
I like this first option here, and moving the `ResponseDomain` to `domains-table` and using select there seems right.